### PR TITLE
[Console] Command lazy loading

### LINKED ISSUE #8147: [Console] Add documentation about the console.command "command" attribute (lazy commands)

### PR DESCRIPTION
**Pull request by chalasr**
Fixes #8147.
Any help regarding wording/formatting would be much appreciated as usual :)

### CODE REVIEW

File: console/lazy_commands.rst [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+
+    $commandLoader = new FactoryCommandLoader(array(
+        'app:heavy' => function () { return new HeavyCommand() },
+    ));
+
+    $application = new Application();
+    $application->setCommandLoader($commandLoader);
+    $application->run();
+
+This way, the ``HeavyCommand`` instance will be created only when the ``app:foo``
```
Comment (yceruto) [inline console/lazy_commands.rst]: ``app:heavy`` instead of ``app:foo``?

File: console/lazy_commands.rst [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+
+    $container = new ContainerBuilder();
+    $container->register(FooCommand::class, FooCommand::class);
+    $container->compile();
+
+    $commandLoader = new ContainerCommandLoader($container, array(
+        'app:foo' => FooCommand::class,
+    ));
+
+Like this, executing the `app:foo` command will load the ``FooCommand`` service
```
Comment (yceruto) [inline console/lazy_commands.rst]: double `` around of "app:foo" ?

**Comment by chalasr** [comment]
@yceruto comments addressed, thanks!

File: console/commands_as_services.rst [hunk 3]
```diff
@@ -66,6 +66,73 @@ works! You can call the ``app:sunshine`` command and start logging.
...
+        protected static $defaultName = 'app:sunshine';
+
+        // ...
+    }
+
+Or set the ``command`` attribute on the ``console.command`` tag in your service definition:
+
+.. configuration-block::
+
+.. code-block:: yaml
```
Comment (xabbuh) [inline console/commands_as_services.rst]: must be indented by four spaces

Comment (chalasr) [reply console/commands_as_services.rst]: fixed

**Review by xabbuh — APPROVED** [review]

**Review by ogizanagi — APPROVED** [review]

**Review by javiereguiluz — APPROVED** [review]

**Comment by javiereguiluz** [comment]
@chalasr thanks for providing this feature and for documenting it. In a minor reword commit, I made this change:

```diff
-register an intermediate router ...
+register an intermediate loader ...
```

Please confirm that it's correct to use "loader" instead of "router". Thanks!

**Comment by chalasr** [comment]
@javiereguiluz Absolutely correct, thank you for that and other your other fixes.

**Comment by weaverryan** [comment]
Thanks @chalasr! Really great feature and accomplishment - congrats :)

**Pull request merged**